src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+ * Dates are aligned such that {@code 0001/01/01 (International fixed)} is {@code 0001-01-01 (ISO)}.</p>
+ * <p>
+ * More information is available in the <a href='https://en.wikipedia.org/wiki/International_Fixed_Calendar'>International Fixed Calendar</a> Wikipedia article.</p>
+ * <p>
+ * <h3>Implementation Requirements</h3>
+ * This class is immutable and thread-safe.
+ * <p>
+ * This class must be treated as a value type. Do not synchronize, rely on the
+ * identity hash code or use the distinction between equals() and ==.</p>
+ */
+public final class InternationalFixedDate
+        extends AbstractDate
+        implements ChronoLocalDate, Serializable {
+    /**
+     * Serialization version UID.
+     */
+    private static final long serialVersionUID = -5501342824322148215L;
+    /**
+     * Leap Day as day-of-year
+     */
+    private static final int LEAP_DAY_AS_DAY_OF_YEAR = 6 * InternationalFixedChronology.DAYS_IN_MONTH + 1;
+    /**
+     * The proleptic year.
+     */
+    private final int prolepticYear;
+    /**
+     * The month of the year.
+     * Leap Day is encoded as month -1, Year Day is encoded as month 0.
+     */
+    private final int month;
+    /**
+     * The day of the month.
+     * Leap Day is encoded as day-of-month -1, Year Day is encoded as day-of-month 0.
+     */
+    private final int day;
+    /**
+     * The day of year.
+     */
+    private final transient int dayOfYear;
+
+    /**
+     * For calculation purposes, internally Leap Day is treated as day-of-year 169,
+     * squeezed between end of month 6 and beginning of month 7.
+     * Similarly, Year Day is treated as day-of-year 365 - or 366 in a leap year.
+     */
+
+    //-----------------------------------------------------------------------
+
+    /**
+     * Creates an instance from validated data.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param month         the International fixed month, from -1 to 13 (-1 for Leap Day, 0 for Year Day)
+     * @param dayOfMonth    the International fixed day-of-month, from -1 to 28 (-1 for Leap Day, 0 for Year Day)
+     * @return the International fixed date
+     */
+    private InternationalFixedDate(final int prolepticYear, final int month, final int dayOfMonth) {
+        this.prolepticYear = prolepticYear;
+        this.month = month;
+        this.day = dayOfMonth;
+        this.dayOfYear = day == -1 ? LEAP_DAY_AS_DAY_OF_YEAR : day == 0 ? InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear() ? 1 : 0) :
+                (month - 1) * InternationalFixedChronology.DAYS_IN_MONTH + day + (isLeapYear() && month > 6 ? 1 : 0);
+    }
+
+    /**
+     * Creates an instance from validated data.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param dayOfYear     the day of the year
+     * @return the International fixed date
+     */
+    private InternationalFixedDate(final int prolepticYear, final int dayOfYear) {
```

Comment:
Hmm, I think I prefer ditching the year-Day constructor, in preference for moving those calculations to `ofYearDay` (and routing through the year/month/day constructor).  Your day-of-year calculation is in two places.